### PR TITLE
Tests/frontend test setup

### PR DESCRIPTION
**Pull request by alvyjudy**
# Commands

- `npm run test` to run all tests in the command line.

- `npm run testInteractive` to open the graphical interface to run tests.]

# Requirement

- both frontend and backend server need to be up and running. Cypress looks for the baseUrl address set up in `cypress.json` (currently, it is set to `http://localhost:3000`)

***

Before I go any deeper, I like to get a sense of what everyone thinks of the front end set up and some decisions I made.

- I used Cypress to test the frontend part of the application because among all the UI testing library, I find it best at emulating actual usage of the app. 
- I intend to stub most of the network request for the tests. Originally I wanted to do a full integration setup, but due to our tight integration with Google API, that doesn't seem to be a happy path. 

Here's how example of how it works, in case cypress is new to some:
```javascript
// loginLogout.spec.js

const email = Cypress.env("email"); // uesttser@gmail.com
describe("Display login, sign up pages", ()=>{
  beforeEach(()=>{
      cy.interceptApiAuthTestWithFailureResponse(); // custom command that intercepts the request to /api/authentication/test and returns a 401 response
      cy.visit("/login"); // the test runner visit the path, prepended by baseUrl. In this case, it visits http://localhost:3000/login
  })

  it("enter email and click button", ()=>{
      cy.get("#email").type(email); // type in the box
      cy.contains("Continue").click(); // click the button
      cy.contains(email);  //assert that it displayed
  })
}

```

In action:

https://user-images.githubusercontent.com/53921639/106642587-8a7d6a80-6556-11eb-8cc8-60e50b7d1840.mov

### CODE REVIEW

**Comment by dproc96** [comment]
Can you update the package.json so this runs on `npm test`?

**Comment by alvyjudy** [comment]
Frontend set up is now complete. This PR also includes the tests for log in log out flow.

File: client/cypress/fixtures/userData.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name":"alveoli",
+  "email":"uesttser@gmail.com",
+  "photoUrl":"https://nypost.com/wp-content/uploads/sites/2/2020/06/spongebob-gay-1.jpg?quality=90&strip=all&w=1236&h=820&crop=1",
```
Comment (WhoaNellie) [inline client/cypress/fixtures/userData.json:4]: 😆

**Review by WhoaNellie — APPROVED** [review]
This is my first time seeing cypress, but I like it! Do you have any examples of what a failed test would look like?

**Review by dproc96 — APPROVED** [review]
Looks great!

**Pull request merged**